File: go/metered_entitlement_value.go
// this file is @generated
package meteroid

type MeteredEntitlementValue struct {
	// Per-entitlement kill switch. `false` means disabled.
	Enabled *bool `json:"enabled,omitempty"`

	// Cap on usage. Null means unlimited.
	Limit *string `json:"limit,omitempty"`

	ResetPeriod *ResetPeriod `json:"reset_period,omitempty"`
}

File: python/meteroid/models/metered_entitlement_value.py
# this file is @generated
import dataclasses
import typing as t
from decimal import Decimal

from ..serialization import BaseModel
from .reset_period import ResetPeriod


@dataclasses.dataclass
class MeteredEntitlementValue(BaseModel):
    enabled: t.Optional[bool] = None
    """Per-entitlement kill switch. `false` means disabled."""

    limit: t.Optional[Decimal] = None
    """Cap on usage. Null means unlimited."""

    reset_period: t.Optional[ResetPeriod] = None
